tests/e2e/framework/deployable_config.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2017,2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package framework
+
+import (
+	"time"
+
+	"go.uber.org/multierr"
+
+	"istio.io/istio/tests/util"
+)
+
+const defaultPropagationDelay = 10 * time.Second
+
+// DeployableConfig is a collection of configs that are applied/deleted as a single unit.
+type DeployableConfig struct {
+	Namespace  string
+	YamlFiles  []string
+	applied    []string
+	Kubeconfig string
+}
+
+// Setup pushes the config and waits for it to propagate to all nodes in the cluster.
+func (c *DeployableConfig) Setup() error {
+	c.applied = []string{}
```

Comment:
Please add an env variable ( or flag, but rather env ) to 'skip_setup' and 'skip_cleanup' - we have a global flag for skipping istio setup as well. 

Will help a lot for debugging individual test cases - you can run a test with the env set, and debug manually after the test fails. Or repeat the test - without the delay.